client/src/controllers/PreviewPanelController.ts

```diff
@@ -0,0 +1,412 @@
+import { Controller } from '@hotwired/stimulus';
+
+import { WAGTAIL_CONFIG } from '../config/wagtailConfig';
+import { gettext } from '../utils/gettext';
+
+/**
+ * Controls the preview panel component to submit the current form state and
+ * update the preview iframe if the form is valid.
+ */
+export class PreviewPanelController extends Controller<HTMLElement> {
+  static classes = ['hidden', 'unavailable', 'hasErrors', 'selectedSize'];
+
+  static targets = [
+    'sizeInput',
+    'newTabButton',
+    'spinner',
+    'modeSelect',
+    'iframe',
+  ];
+
+  static values = {
+    baseUrl: String,
+    autoUpdate: Boolean,
+    autoUpdateInterval: Number,
+  };
```

Comment:
```suggestion
  static values = {
    baseUrl: {default: '', type: String},
    autoUpdate: {default: false, type: Boolean},
    autoUpdateInterval: {default: 50, type: Number,
  };
```

I recommend we put default values for these, Stimulus will throw an error if you try to access a value that is not set. This may be ok but it's nice to have defaults. It also makes the data attributes not required if you know it's going to work as default (e.g. maybe auto-update interval).

Reply:
Thanks! Apparently Stimulus will _not_ throw an error for a missing Value, and instead will return a default value based on its Getter:

https://stimulus.hotwired.dev/reference/values#getters

I've added the defaults, but I also made it so that an exception is thrown if `baseUrl` is empty, because it's necessary for the preview to work.

Reply:
Yeah. If you don't set a default it will through an error if you attempt to access. Up do you on whether you want to have a default and throw your own error though.